### PR TITLE
[Issue #10540] Add shared schema types for non-negative monetary and integer percentage

### LINKED ISSUE #10540: Add shared schema types for non-negative monetary and integer percentage
### Summary

SF424C requires three new shared JSON schema types: non negative monetary inputs (no leading minus), non negative monetary totals, and a whole number percentage (integer 0-100). These can't be expressed by tweaking existing types because budget_monetary_amount allows negatives, and we have no percentage type at all. Staging these as a standalone ticket (separate from the form build) so the percentage contract gets reviewed alongside the arithmetic rule that depends on it.

### Acceptance criteria

- [x] 3 shared data types created
  - [x] `budget_monetary_amount_non_negative` for user-input monetary cells.
  - [x] `budget_monetary_total_non_negative` for computed subtotal/total cells.
  - [x] `percentage`integer, minimum: 0, maximum: 100. Docstring must spell out the contract: 5 means 5%, not 0.05. Used as the right-hand operand in `multiply_by_percentage` (added in a separate ticket).
- [x] All three types referenceable via `COMMON_SHARED_V1.field_ref("...")`.
- [x] Unit tests cover: valid inputs accepted; negative monetary inputs rejected; decimal percentage rejected; out of range percentage rejected; oversized monetary rejected.
- [x] No changes to any existing form's schema.

### PR DESCRIPTION
**Pull request by mikehgrantsgov**
## Summary

Fixes #10540

## Changes proposed

`budget_monetary_amount_non_negative` - string type with same regex as budget_monetary_amount but leading minus sign disallowed
`budget_monetary_total_non_negative` - separate name to distinguish computed subtotal/total cells from user-input cells
`percentage` - intended as the right-hand operand for multiply_by_percentage (added in a separate ticket)

Add unit tests for all three types covering valid inputs, negative rejection, invalid patterns, min/max length (monetary), and out-of-range/decimal rejection (percentage)

## Context for reviewers

The existing `budget_monetary_amount` allows negatives, so the non-negative types can't be expressed by tweaking it — they need separate entries.

No existing form schemas were modified.

SF424C requires three new shared JSON schema types: non negative monetary inputs (no leading minus), non negative monetary totals, and a whole number percentage (integer 0-100). These can't be expressed by tweaking existing types because budget_monetary_amount allows negatives, and we have no percentage type at all. Staging these as a standalone ticket (separate from the form build) so the percentage contract gets reviewed alongside the arithmetic rule that depends on it.

## Validation steps

See new unit tests added.

### CODE REVIEW

File: api/tests/src/form_schema/shared/test_common_shared.py [hunk 1]
```diff
@@ -352,3 +352,131 @@ def test_shared_common_v1_sam_uei_max_length():
     assert len(validation_issues) == 1
     assert validation_issues[0].type == "maxLength"
     assert validation_issues[0].field == "$.my_field"
+
+
+###################################
+# Budget Monetary Amount Non-Negative
+###################################
```
Comment (joshtonava) [inline api/tests/src/form_schema/shared/test_common_shared.py:359]: **suggestion:** consider adding the , `min_length` (empty string → `minLength`) and `invalid_pattern` cases that the other similar amount type has.

**Review by joshtonava — DISMISSED** [review]

**Review by joshtonava — APPROVED** [review]
